requirements.txt

```diff
@@ -19,6 +19,7 @@ idna==3.4
 imagesize==1.4.1
 Jinja2==3.1.2
 MarkupSafe==2.1.3
+matplotlib==3.7.2
```

Comment:
This file is generated and should not be edited. I add the base dependencies here: https://github.com/oneapi-src/distributed-ranges/blob/cb32defeff7893c10e360f7af50c41eec8c9a350/base-requirements.txt#L1
and do:
```
pip install -r base-requirements.txt
pip freeze > requirements.txt
```
and then add the copyright header to the top of requirements.txt. I did it this way because otherwise it is hard to tell if packages are no longer needed. I didn't do any research in the best way to manage the requirements.txt so there might be a better way to do it.

Reply:
I don't see these two lines (pip install, pip freeze) in our repo. How do you call them?
Is this a manual procedure of adding a dependency:
1. create venv
2. install packages from current requirements.txt: pip install -r requirements.txt
3. edit base-requirements.txt
4. do pip install -r base-requirements.txt
5. do pip freeze > requirements.txt
6. commit changes to repo
right?

Reply:
Yes. I do it manually.

Reply:
I have done it the same way now but maybe after rebasing you can see 'my' changes. It was not done manually.